core/controllers/base.py

```diff
@@ -358,7 +358,14 @@ def vaidate_and_normalize_args(self):
                     handler_args.update(payload_args)
             else:
                 request_arg_keys.append(arg)
-                handler_args[arg] = self.request.get(arg)
+                param = self.request.get(arg)
```

Comment:
I do not think we should do this. I have left a rationale in that comment.

Reply:
Oops, sorry, I think I got confused by what @Nik-09 meant by "schema validation system".

I think it's correct for this particular normalization to happen outside schema_utils (but maybe allow case-insensitivity of the string, i.e. convert 'tRue' to True as well). However -- shouldn't we check that the expected param is of type boolean before doing this?

Reply:
@Nik-09 -- did you miss this part?

> However -- shouldn't we check that the expected param is of type boolean before doing this?

Reply:
Thank you @seanlip.
There is a pros & cons for wirting this normalization in base.py or payload_validator.py.
**Pros for adding in base.py**
1. We will strictly apply this normalization for only GET/DELETE request method, since they only face the arguments of expected type bool but received string.

**Cons for adding in base.py**

1. We have to rearrange the code to extract schema first then check whether the expected type is bool, if yes then only apply normalization.
    - Currently schema is extracted after extracting all the params from request.
    - It will deviate from our idea since we are normalizing in base.py but expected normalization to be performed in payload_validator.

**Pros for adding in payload_validator**

Reply:
Your last section is empty, but it seems reasonable to me to do the normalization in payload_validator given that we need to know what we're normalizing it to. Does this break anything?

Reply:
Sorry for sending initially partial message.
Now its complete. Please take a look.
Thank you.

Reply:
> Does this break anything?

I have tested it for editor.py and now testing/playing with other features. It seems to work perfectly.

Reply:
Can the payload validator check what sort of method it is received from -- does it have any way to know?

Regardless of the answer, I feel like the right place for this is probably the payload validator. But it would be better if it had knowledge of the context for whether this conversion is allowed. Maybe you could pass in an argument for whether to allow string to bool conversion.

Reply:
Or, another thought, what calls the payload_validator method? Possibly another approach is to do the conversion just before that method is called, so that that method doesn't need to know the extra context.

Reply:
I also feel that the normalization must be in payload_validator.
For sencond comment -- base.py is the place from where payload_validator is called for validation & normalization.
Thus its again the same case as previous attempt.

I have adopted the solution of pass extra argument ```allow_string_to_bool_conversion```
Now this pr is ready for full pass. Please take a look.
Thank you.